Makefile

```
@@ -1,17 +1,74 @@
-.PHONY: build-deps test-deps run-tests update-tests
+RATE_TIME_RANGE ?= 2m
+POLICY ?= autogen
+OUTPUT_STATIC_DASHBOARD ?= dashboard.json
 
+.PHONY: build-deps
 build-deps:
 	go get github.com/google/go-jsonnet/cmd/jsonnet
 	go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
 	jb install
 
+
+_build-static-prometheus:
+ifndef DATASOURCE
+	@echo 1>&2 "DATASOURCE must be set"
+		false
+endif
+ifndef JOB
+	@echo 1>&2 "JOB must be set"
+		false
+endif
+	# RATE_TIME_RANGE is optional, default is "2m"
+	jsonnet -J ./vendor -J . \
+		--ext-str DATASOURCE=${DATASOURCE} \
+		--ext-str JOB=${JOB} \
+		--ext-str RATE_TIME_RANGE=${RATE_TIME_RANGE} \
+		dashboard/build/prometheus/${DASHBOARD_BUILD_SOURCE} -o ${OUTPUT_STATIC_DASHBOARD}
+
+.PHONY: build-static-prometheus
+build-static-prometheus:
+	make DASHBOARD_BUILD_SOURCE=dashboard_static.jsonnet _build-static-prometheus
```

For all `make` in the file:
```suggestion
	$(MAKE) DASHBOARD_BUILD_SOURCE=dashboard_static.jsonnet _build-static-prometheus
```
https://www.gnu.org/software/make/manual/make.html#MAKE-Variable

Also you could try to do something like that:

```Makefile
_source-static: DASHBOARD_BUILD_SOURCE=dashboard_static.jsonnet
_source-tdg-static: DASHBOARD_BUILD_SOURCE=tdg_dashboard_static.jsonnet
_dashboard_prometheus: DASHBOARD=dashboard/build/prometheus/$(DASHBOARD_BUILD_SOURCE)
_dashboard_influxdb: DASHBOARD=dashboard/build/influxdb/$(DASHBOARD_BUILD_SOURCE)

_build-static:
	bla-bla-bla
	jsonnet -J ./vendor -J . \
		--ext-str DATASOURCE=${DATASOURCE} \
		--ext-str JOB=${JOB} \
		--ext-str RATE_TIME_RANGE=${RATE_TIME_RANGE} \
		$(DASHBOARD) -o $(OUTPUT_STATIC_DASHBOARD)

build-static-prometheus: _source-static _dashboard_prometheus _build-static
build-static-tdg-prometheus: _source-tdg-static _dashboard_prometheus _build-static
build-static-influxdb: _source-static _dashboard_influxdb _build-static
build-static-tdg-influxdb: _source-tdg-static _dashboard_influxdb _build-static
...
```

It may look more clean.

---

Thank you! I'm not well-experienced in `make` and that point is bothered me. Used the first solution, the second one didn't worked for me (target failed with `No rule to make target '_build-static-source'`).